src/Moq/Mock.cs

```diff
@@ -389,6 +389,24 @@ internal static void VerifyNoOtherCalls(Mock mock)
 				l.Body is MethodCallExpression lc && r.Body is MethodCallExpression rc && lc.Method == rc.Method;
 		}
 
+		internal void AddFixedReturnValueSetup(MethodInfo method, IReadOnlyList<Expression> arguments, LambdaExpression expression, object returnValue)
+		{
+			Debug.Assert(Mock.TryGetFromReturnValue(returnValue, out _));
+
+			if (expression.IsProperty())
+			{
+				var property = expression.ToPropertyInfo();
+				Guard.CanRead(property);
+
+				Debug.Assert(method == property.GetGetMethod(true));
+			}
+
+			ThrowIfSetupExpressionInvolvesUnsupportedMember(expression, method);
+			ThrowIfSetupMethodNotVisibleToProxyFactory(method);
```

Comment:
Strange that we need validation only in this new overload, but not in the preexisting one...?